src/libraries/System.Private.CoreLib/src/System/Environment.Linux.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static ReusableTextReader reusableReader = new ReusableTextReader();
+        public static long WorkingSet => (long)(Interop.procfs.TryReadStatusFile(ProcessId, out Interop.procfs.ParsedStatus status, reusableReader) ? status.VmRSS : 0);
```

Comment:
Noticed that the return value isn't checked in ProcessManager.Linux.cs, only in Process.Linux.cs. The check here doesn't achieve anything as it will give back a struct with defaults if it fails, but it's harmless as well.